src/factory-in-proc.js

```diff
@@ -101,9 +101,12 @@ class FactoryInProc {
       ? options.init
       : true
 
-    if (!options.disposable) {
+    if (options.disposable) {
+      options.config = defaults({}, options.config, defaultConfig)
+    } else {
       const nonDisposableConfig = clone(defaultConfig)
-      delete nonDisposableConfig.Addresses
+      // TODO why delete the addrs here???
+      // delete nonDisposableConfig.Addresses
```

Comment:
@dryajov why were you deleting the addrs here?

Reply:
Because at some point we wanted non-disposable nodes to use `js-ipfs` or `go-ipfs` default ports and not random ports.